packages/changelog/src/index.ts

```diff
@@ -171,67 +171,127 @@ const applyChanges = ({
-          indentation: number,
-          indentFirstLine = false
+          indentation: number = 4
         ) => {
-          // indent the summary so it fits under the bullet point
+          // indent the summary so it fits under the new bullet point we added
           const summary = release.summary.replace(
             /\n/g,
-            `\n${"  ".repeat(indentFirstLine ? indentation + 2 : indentation)}`
+            `\n${" ".repeat(indentation)}`
           );
-          const firstLinIndent = indentFirstLine
-            ? "  ".repeat(indentation)
-            : "";
 
           if (!release.meta || (!!release.meta && !release.meta.commits)) {
-            addition += `\n${firstLinIndent}- ${summary}`;
+            addition += `\n- ${summary}`;
           } else {
             const commit = release.meta.commits![0];
-
             const commitLink = `[\`${commit.hashShort}\`](${gitSiteUrl}commit/${commit.hashLong})`;
-
             const [, pr] = /(#[0-9]+)/g.exec(commit.commitSubject) ?? [];
             const prLink = pr
               ? `([${pr}](${gitSiteUrl}pull/${pr.slice(1)}))`
               : "";
 
-            addition += `\n${firstLinIndent}- ${commitLink}${prLink} ${summary}`;
+            addition += `\n- ${commitLink}${prLink} ${summary}`;
           }
         };
 
-        for (const release of assembledChanges.releases[
-          change.changes.name
-        ].changes.filter((c) => !c.meta?.dependencies)) {
-          renderRelease(release, 1);
-        }
-
-        const groupedDpesChanges: {
+        /**
+         * Untagged changes are changes that don't have a tag associated with and `config.defaultChangeTag` is not set.
+         * These will be rendered without a tag (section or category) at the beginning of a release changelog.
+         */
+        const untaggedChanges: {
+          summary: string;
+          meta?: Meta | undefined;
+          tag?: string | undefined;
+        }[] = [];
+        const groupedChangesByTag: {
           [k: string]: {
             summary: string;
             meta?: Meta | undefined;
+            tag?: string | undefined;
           }[];
         } = {};
 
+        // **IMPORTANT**: prefill the `groupedChangesByTag` with tags from config
+        // in the same order they were defined
+        Object.keys(config.changeTags ?? {}).forEach(
+          (k) => (groupedChangesByTag[k] = [])
+        );
+
         assembledChanges.releases[change.changes.name].changes
-          .filter((c) => c.meta?.dependencies)
-          .forEach((r) => {
-            r.meta!.dependencies.forEach((key) => {
-              if (!groupedDpesChanges[key]) groupedDpesChanges[key] = [];
-              groupedDpesChanges[key].push(r);
-            });
+          .filter((c) => !c.meta?.dependencies)
+          .forEach((c) => {
+            // fallback to `defaultChangeTag` if it is set, otherwise mark this change as untagged
+            const key = c.tag ?? config.defaultChangeTag;
```

Comment:
I guess I was presuming to put it in the user-defined tags if that makes sense. It feels odd to me that someone would specifically add a `default` without having it be the default if something isn't specified. Perhaps that would be expecting that config to handle too much / have too many caveats there?

Reply:
I am afraid that using `default` inside `tags` will confuse users since `tags` is a mapping object not configuration object. I feel like my proposed solution above is a good in-between.

Reply:
Yea, you are probably right. Let's keep it as you have it now with `defaultChangeTag`.

Reply:
what about this one https://github.com/jbolda/covector/pull/244#discussion_r1178053226 ? I think it has good grouping and still won't confuse users